### PR TITLE
PLT-2037 - Adding input validation guidelines

### PR DESCRIPTION
**Pull request by asaadmahmoodspin**
Hey @esethna have a look.

### CODE REVIEW

File: source/developer/fx-guidelines.rst [hunk 1]
```diff
@@ -45,16 +45,16 @@ Forgiving extends to compatibility as well, and whether you’re working from a
...
+Bootstrap classes
 -----------------------------------
 
 Since we’re using [bootstrap](<http://getbootstrap.com/), try to utilize [bootstrap classes](<http://getbootstrap.com/css/>) as much as possible rather than creating new styles or using inline styles.
 
-Custom Styles
+Custom styles
 -----------------------------------
 
 We’re using [BEM methodology](https://en.bem.info/method/) and would advise everyone to do so when creating new classes or ids. There’s a lot of places in our app that currently don’t follow BEM but we’re trying to improve the codebase. [Here](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/) is a brief overview of the BEM syntax.
```
Comment (esethna) [inline source/developer/fx-guidelines.rst]: These URLs don't seem to be hyperlinking in the preview or on docs.mattermost?

File: source/developer/fx-guidelines.rst [hunk 2]
```diff
@@ -8,7 +8,7 @@ The purpose of the Mattermost Fan eXperience Guidelines ("FX Guidelines") is to
 
 |
 
-Design Principles
+Design principles
```
Comment (esethna) [inline source/developer/fx-guidelines.rst]: Not sure why we are changing to sentence case for these headings? I think we decided in the docs guidelines headings of level H1, H2, H3 should be title case? http://docs.mattermost.com/process/documentation-guidelines.html#use-appropriate-heading-case

File: source/developer/fx-guidelines.rst [hunk 3]
```diff
@@ -45,16 +45,16 @@ Forgiving extends to compatibility as well, and whether you’re working from a
```
Comment (esethna) [inline source/developer/fx-guidelines.rst]: "Fast has two parts: being _responsive_ and being _quick_" doesn't seem to be rendering on github or docs.mattermost.

**Comment by esethna** [comment]
Thanks for the tweaks and additions Asaad! I've made some comments^

**Comment by asaadmahmoodspin** [comment]
Hey @esethna did the improvements, have a look.

**Comment by esethna** [comment]
@asaadmahmoodspin ^

**Comment by asaadmahmoodspin** [comment]
Sure.

**Comment by esethna** [comment]
Thanks Asaad, merged!

**Pull request merged**